### DISCUSSION TITLE
Unable to configure graph.set-vert-id for CQL use

### DISCUSSION THREAD

**Discussion by ZhangErling** [JanusGraph/janusgraph#3012] [Q&A] [answered]
I learned this method and set the custom ID
https://github.com/JanusGraph/janusgraph/issues/2051#issuecomment-601789026

> gremlin> graph = JanusGraphFactory.open('conf/janusgraph-inmemory.properties')
==>standardjanusgraph[inmemory:[127.0.0.1]]
gremlin> g = graph.traversal()
==>graphtraversalsource[standardjanusgraph[inmemory:[127.0.0.1]], standard]
gremlin> g.addV("person").property(T.id, 8).next()
==>v[8]
gremlin>

Thank you for reading, any answers or suggestions will help me！

Failed when I set the custom ID for CQL

Properties files are configured graph.set-vert-id =true, The other configurations are the default ones.

**Plan 1**

> gremlin> graph = JanusGraphFactory.open('conf/janusgraph-cql.properties')
10:48:39 WARN  org.janusgraph.diskstorage.configuration.builder.ReadConfigurationBuilder  - Local setting graph.set-vertex-id=true (Type: FIXED) is overridden by globally managed value (false).  Use the ManagementSystem interface instead of the local configuration to control this setting.
10:48:39 WARN  org.janusgraph.diskstorage.configuration.builder.ReadConfigurationBuilder  - Local setting cache.db-cache-clean-wait=20 (Type: GLOBAL_OFFLINE) is overridden by globally managed value (50).  Use the ManagementSystem interface instead of the local configuration to control this setting.
10:48:39 WARN  org.janusgraph.diskstorage.configuration.builder.ReadConfigurationBuilder  - Local setting cache.db-cache-time=180000 (Type: GLOBAL_OFFLINE) is overridden by globally managed value (10000).  Use the ManagementSystem interface instead of the local configuration to control this setting.
==>standardjanusgraph[cql:[127.0.0.1]]
gremlin> g = graph.traversal()
==>graphtraversalsource[standardjanusgraph[cql:[127.0.0.1]], standard]
gremlin> g.addV("person").property(T.id, 8).next()
Vertex does not support user supplied identifiers
Type ':help' or ':h' for help.
Display stack trace? [yN]
gremlin>

**Plan 2**

1. server

> 2    [main] INFO  org.apache.tinkerpop.gremlin.server.GremlinServer  - 1.21.0
         \,,,/
         (o o)
-----oOOo-(3)-oOOo-----
>
>134  [main] INFO  org.apache.tinkerpop.gremlin.server.GremlinServer  - Configuring Gremlin Server from D:\tools\janusgraph-0.6.1\conf\gremlin-server\gremlin-server.yaml
181  [main] INFO  org.apache.tinkerpop.gremlin.server.util.MetricManager  - Configured Metrics ConsoleReporter configured with report interval=180000ms
186  [main] INFO  org.apache.tinkerpop.gremlin.server.util.MetricManager  - Configured Metrics CsvReporter configured with report interval=180000ms to fileName=/tmp/gremlin-server-metrics.csv
246  [main] INFO  org.apache.tinkerpop.gremlin.server.util.MetricManager  - Configured Metrics JmxReporter configured with domain= and agentId=
251  [main] INFO  org.apache.tinkerpop.gremlin.server.util.MetricManager  - Configured Metrics Slf4jReporter configured with interval=180000ms and loggerName=org.apache.tinkerpop.gremlin.server.Settings$Slf4jReporterMetrics
526  [main] INFO  org.apache.commons.beanutils.FluentPropertyBeanIntrospector  - Error when creating PropertyDescriptor for public final void org.apache.commons.configuration2.AbstractConfiguration.setProperty(java.lang.String,java.lang.Object)! Ignoring this property.
1118 [main] INFO  com.datastax.oss.driver.internal.core.DefaultMavenCoordinates  - DataStax Java driver for Apache Cassandra(R) (com.datastax.oss:java-driver-core) version 4.13.0
1593 [JanusGraph Session-admin-0] INFO  com.datastax.oss.driver.internal.core.time.Clock  - Using native clock for microsecond precision
2221 [main] INFO  org.janusgraph.diskstorage.configuration.ExecutorServiceBuilder  - Initiated fixed thread pool of size 10
2538 [main] WARN  org.janusgraph.diskstorage.configuration.builder.ReadConfigurationBuilder  - **Local setting graph.set-vertex-id=true (Type: FIXED) is overridden by globally managed value (false).  Use the ManagementSystem interface instead of the local configuration to control this setting.**
2552 [main] WARN  org.janusgraph.diskstorage.configuration.builder.ReadConfigurationBuilder  - Local setting cache.db-cache-clean-wait=20 (Type: GLOBAL_OFFLINE) is overridden by globally managed value (50).  Use the ManagementSystem interface instead of the local configuration to control this setting.
2568 [main] WARN  org.janusgraph.diskstorage.configuration.builder.ReadConfigurationBuilder  - Local setting cache.db-cache-time=180000 (Type: GLOBAL_OFFLINE) is overridden by globally managed value (10000).  Use the ManagementSystem interface instead of the local configuration to control this setting.
2605 [main] INFO  org.janusgraph.graphdb.idmanagement.UniqueInstanceIdRetriever  - Generated unique-instance-id=c0a889013776-RYZEN7-4800H1
2648 [JanusGraph Session-admin-0] INFO  com.datastax.oss.driver.internal.core.time.Clock  - Using native clock for microsecond precision
2823 [main] INFO  org.janusgraph.diskstorage.configuration.ExecutorServiceBuilder  - Initiated fixed thread pool of size 10
2828 [main] INFO  org.janusgraph.diskstorage.configuration.ExecutorServiceBuilder  - Initiated fixed thread pool of size 32
3213 [main] INFO  org.janusgraph.diskstorage.Backend  - Configuring total store cache size: 203604752
3637 [main] INFO  org.janusgraph.diskstorage.log.kcvs.KCVSLog  - Loaded unidentified ReadMarker start time 2022-04-16T03:05:50.837Z into org.janusgraph.diskstorage.log.kcvs.KCVSLog$MessagePuller@38704ff0
3640 [main] INFO  org.apache.tinkerpop.gremlin.server.util.ServerGremlinExecutor  - Initialized Gremlin thread pool.  Threads in pool named with pattern gremlin-*
3679 [main] INFO  org.apache.tinkerpop.gremlin.server.util.ServerGremlinExecutor  - Initialized GremlinExecutor and preparing GremlinScriptEngines instances.
6205 [main] INFO  org.apache.tinkerpop.gremlin.server.util.ServerGremlinExecutor  - Initialized gremlin-groovy GremlinScriptEngine and registered metrics
6209 [main] INFO  org.apache.tinkerpop.gremlin.server.util.ServerGremlinExecutor  - A GraphTraversalSource is now bound to [g] with graphtraversalsource[standardjanusgraph[cql:[127.0.0.1]], standard]
6231 [main] INFO  org.apache.tinkerpop.gremlin.server.op.OpLoader  - Adding the standard OpProcessor.
6306 [main] INFO  org.apache.tinkerpop.gremlin.server.op.OpLoader  - Adding the session OpProcessor.
6310 [main] INFO  org.apache.tinkerpop.gremlin.server.op.OpLoader  - Adding the traversal OpProcessor.
6314 [main] INFO  org.apache.tinkerpop.gremlin.server.GremlinServer  - Executing start up LifeCycleHook
6330 [main] INFO  org.apache.tinkerpop.gremlin.server.GremlinServer  - Executed once at startup of Gremlin Server.
6342 [main] INFO  org.apache.tinkerpop.gremlin.server.GremlinServer  - idleConnectionTimeout was set to 0 which resolves to 0 seconds when configuring this value - this feature will be disabled
6342 [main] INFO  org.apache.tinkerpop.gremlin.server.GremlinServer  - keepAliveInterval was set to 0 which resolves to 0 seconds when configuring this value - this feature will be disabled
6361 [main] INFO  org.apache.tinkerpop.gremlin.server.AbstractChannelizer  - Configured application/vnd.graphbinary-v1.0 with org.apache.tinkerpop.gremlin.driver.ser.GraphBinaryMessageSerializerV1
6361 [main] INFO  org.apache.tinkerpop.gremlin.server.AbstractChannelizer  - Configured application/vnd.graphbinary-v1.0-stringd with org.apache.tinkerpop.gremlin.driver.ser.GraphBinaryMessageSerializerV1
6364 [main] WARN  org.apache.tinkerpop.gremlin.server.AbstractChannelizer  - The org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV3d0 serialization class is deprecated.
6439 [main] INFO  org.apache.tinkerpop.gremlin.server.AbstractChannelizer  - Configured application/vnd.gremlin-v3.0+gryo with org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV3d0
6439 [main] WARN  org.apache.tinkerpop.gremlin.server.AbstractChannelizer  - The org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV3d0 serialization class is deprecated.
6440 [main] INFO  org.apache.tinkerpop.gremlin.server.AbstractChannelizer  - Configured application/vnd.gremlin-v3.0+gryo-stringd with org.apache.tinkerpop.gremlin.driver.ser.GryoMessageSerializerV3d0
6456 [main] INFO  org.apache.tinkerpop.gremlin.server.AbstractChannelizer  - Configured application/vnd.gremlin-v3.0+json with org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerV3d0
6456 [main] INFO  org.apache.tinkerpop.gremlin.server.AbstractChannelizer  - Configured application/json with org.apache.tinkerpop.gremlin.driver.ser.GraphSONMessageSerializerV3d0
6479 [gremlin-server-boss-1] INFO  org.apache.tinkerpop.gremlin.server.GremlinServer  - Gremlin Server configured with worker thread pool of 1, gremlin pool of 16 and boss thread pool of 1.
6479 [gremlin-server-boss-1] INFO  org.apache.tinkerpop.gremlin.server.GremlinServer  - Channel started at port 8182.

2.  client
> gremlin> :remote connect tinkerpop.server conf/remote-objects.yaml session
==>Configured localhost/127.0.0.1:8182-[a9af72b6-0086-4e6f-97b4-1242d5fddb85]
gremlin>
gremlin> :remote console
==>All scripts will now be sent to Gremlin Server - [localhost/127.0.0.1:8182]-[a9af72b6-0086-4e6f-97b4-1242d5fddb85] - type ':remote console' to return to local mode
gremlin> g.addV("person").property(T.id, 8).next()
Vertex does not support user supplied identifiers
Type ':help' or ':h' for help.
Display stack trace? [yN]
gremlin>

3.  log

> 1024636 [gremlin-server-session-1] WARN  org.apache.tinkerpop.gremlin.server.op.AbstractEvalOpProcessor  - Exception processing a script on request [RequestMessage{, requestId=ff0cf96c-ccbe-472f-a0f8-6b91500ec426, op='eval', processor='session', args={gremlin=g.addV("person").property(T.id, 8).next(), session=a9af72b6-0086-4e6f-97b4-1242d5fddb85, manageTransaction=false, userAgent=Gremlin Console/1.21.0, batchSize=64, maintainStateAfterException=false}}].
java.lang.UnsupportedOperationException: Vertex does not support user supplied identifiers
        at org.apache.tinkerpop.gremlin.structure.Vertex$Exceptions.userSuppliedIdsNotSupported(Vertex.java:163)
        at org.janusgraph.graphdb.tinkerpop.JanusGraphBlueprintsTransaction.addVertex(JanusGraphBlueprintsTransaction.java:111)
        at org.janusgraph.graphdb.tinkerpop.JanusGraphBlueprintsGraph.addVertex(JanusGraphBlueprintsGraph.java:143)
        at org.janusgraph.graphdb.tinkerpop.JanusGraphBlueprintsGraph.addVertex(JanusGraphBlueprintsGraph.java:59)
        at org.apache.tinkerpop.gremlin.process.traversal.step.map.AddVertexStartStep.processNextStart(AddVertexStartStep.java:102)
        at org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.next(AbstractStep.java:135)
        at org.apache.tinkerpop.gremlin.process.traversal.step.util.AbstractStep.next(AbstractStep.java:40)
        at org.apache.tinkerpop.gremlin.process.traversal.util.DefaultTraversal.next(DefaultTraversal.java:230)
        at java_util_Iterator$next.call(Unknown Source)
        at org.codehaus.groovy.runtime.callsite.CallSiteArray.defaultCall(CallSiteArray.java:47)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:115)
        at org.codehaus.groovy.runtime.callsite.AbstractCallSite.call(AbstractCallSite.java:119)
        at Script4.run(Script4.groovy:1)
        at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.eval(GremlinGroovyScriptEngine.java:676)
        at org.apache.tinkerpop.gremlin.groovy.jsr223.GremlinGroovyScriptEngine.eval(GremlinGroovyScriptEngine.java:378)
        at javax.script.AbstractScriptEngine.eval(Unknown Source)
        at org.apache.tinkerpop.gremlin.groovy.engine.GremlinExecutor.lambda$eval$0(GremlinExecutor.java:272)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
        at java.util.concurrent.FutureTask.run(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.lang.Thread.run(Unknown Source)

**Plan 3**

1. server
> **Plan 2**
>
>1. server

2.client

> gremlin> g = traversal().withRemote(DriverRemoteConnection.using("localhost", 8182));
==>graphtraversalsource[emptygraph[empty], standard]
gremlin> g.addV("person").property(T.id, 8).next()
org.apache.tinkerpop.gremlin.driver.exception.ResponseException: Vertex does not support user supplied identifiers
Type ':help' or ':h' for help.
Display stack trace? [yN]
gremlin>

**Comment by li-boxuan** [comment, marked as answer]
`graph.set-vert-id =true` is a fixed configuration, so it can only be set once before you first open the graph. Do you see the same problem when you open a brand new graph?

**Reply by ZhangErling** [reply]
Yes, it must be brand new and never used before it was configured, including **never opened**.

Later, I solved this problem by referring to your answer. Thank you for your answer, and I'm sorry that IT took so long for me to reply to your answer
